File: java/src/main/java/com/dddeurope/niuz/services/AuthorService.java
package com.dddeurope.niuz.services;


import com.dddeurope.niuz.events.ContractSigned;
import com.dddeurope.niuz.events.EventPublisher;
import com.dddeurope.niuz.newsroom.AuthorRepository;

public class AuthorService {
    private final AuthorRepository authors;
    private final EventPublisher publisher;

    public AuthorService(AuthorRepository authors, EventPublisher publisher) {
        this.authors = authors;
        this.publisher = publisher;
    }

    public void hire(String authorId, String authorName, String bankAccount, String contractType, int rate) {
        publisher.publish(new ContractSigned(authorId, contractType, rate, bankAccount, authorName));
    }
}

File: dotnet/niuz.application/services/AuthorService.cs
using niuz.application.events;
using niuz.application.newsroom;

namespace niuz.application.services
{
    public class AuthorService
    {
        private readonly IAuthorRepository authors;
        private readonly IEventPublisher publisher;

        public AuthorService(IAuthorRepository authors, IEventPublisher publisher)
        {
            this.authors = authors;
            this.publisher = publisher;
        }

        public void Hire(string authorId, string authorName, string bankAccount, string contractType, int rate)
        {
            publisher.Publish(new ContractSigned(authorId, authorName, contractType, rate, bankAccount));
        }
    }
}